Dockerfile

```diff
@@ -1,40 +1,28 @@
 FROM node:10.13-alpine
 
-# Install required libraries
-RUN apk --no-cache add bash curl git \
-  make gcc g++ python \
-  libsecret-dev glib-dev
-
-# Set up deploy user and working directory
-RUN adduser -D -g '' deploy
-RUN mkdir -p /app
-RUN chown deploy:deploy /app
-
-# Set up dumb-init
-ADD https://github.com/Yelp/dumb-init/releases/download/v1.2.2/dumb-init_1.2.2_amd64 /usr/local/bin/dumb-init
-RUN chown deploy:deploy /usr/local/bin/dumb-init
-RUN chmod +x /usr/local/bin/dumb-init
-
-# Copy required libs and files to /app
-ADD . /app
-ADD package.json /app
-ADD yarn.lock /app
 WORKDIR /app
 
-# Set up node_modules
-RUN yarn cache clean && yarn install
+RUN apk --no-cache --quiet add \
+      bash \
+      build-base \
+      curl \
+      dumb-init \
+      git \
+      python && \
+      adduser -D -g '' deploy
 
-# Copy over node_modules
-ADD --chown=deploy:deploy . /app
+COPY package.json yarn.lock ./
+COPY patches ./patches
 
-# Compile assets
-RUN yarn assets
-RUN yarn build:server
+RUN yarn install --frozen-lockfile --quiet
+
+COPY . ./
+
+RUN yarn assets && \
```

Comment:
Yes agreed, this could be something more declarative -- maybe `compile-assets`? I'm curious if there's a standard or a better example in another app.

Reply:
Looking at `package.json`, the two lines seem to do the same thing. They both call `yarn webpack` with `NODE_ENV=production`.

Reply:
interesting, @damassi do you know what the `BUILD_SERVER` flag changes here? https://github.com/artsy/force/blob/master/package.json#L13

Reply:
Yup -- it compiles our [server-side code](https://github.com/artsy/force/blob/master/webpack/index.js#L29)